### ISSUE TITLE
Pas de cookie par Google Adsense... Mais fournisseurs partenaires oui ?!

### ISSUE THREAD

**Issue by DominicDesbiens** [AmauriC/tarteaucitron.js#136]
Bonjour,

En ajoutant Google Adsense à la liste des services, il indique ne déposer aucun cookie. Or, en autorisant Adsense, il y a des cookies des fournisseurs partenaires qui s'ont enregistrés. Je parle notamment du cookie doubleclick.net qui est une technologie appartenant à Google, Il n'y a aucune vérification de ceux-ci ?

Merci

**Comment by AmauriC** [comment]
C'est impossible. Le script ne peut lire que les cookies sur son domaine.

Le 5 juin 2018 21:23, DominicDesbiens <notifications@github.com> a écrit :

Bonjour,

En ajoutant Google Adsense à la liste des services, il indique ne déposer aucun cookie. Or, en autorisant Adsense, il y a des cookies des fournisseurs partenaires qui s'ont enregistrés. Je parle notamment du cookie doubleclick.net qui est une technologie appartenant à Google, Il n'y a aucune vérification de ceux-ci ?

Merci

**Comment by DominicDesbiens** [comment]
Merci. Il n'y aurait alors pas possibilité d'ajouter dans le script le nom de cookie déposé par des fournisseurs partenaires de Google ?

Dans l'optique où Adsense ne dépose que les cookies des partenaires, l'afficher dans la liste ne sert finalement qu'à bloquer la génération des cookies des partenaires sans pouvoir compter réellement le nombre de cookies déposés.

**Comment by jipexu** [comment]
> Dans l'optique où Adsense ne dépose que les cookies des partenaires, l'afficher dans la liste ne sert finalement qu'à bloquer la génération des cookies des partenaires sans pouvoir compter réellement le nombre de cookies déposés.

...oui ...
on sait combien de cookies un service est capable de déposé mais on ne peut compter et manipuler que ceux déposé sur le domaine .... et donc effectivement la désactivation du service permet de ne pas utiliser .... donc pas de dépôt .... 
 il faut juste affiner la phrase générée

#121

**Comment by AmauriC** [comment]
Je suis d'accord sur le principe mais ce sera impossible à priori de proposer quelque chose de complet. Adsense peut charger des centaines de régies partenaires, comment savoir qui va utiliser quoi ?

**Comment by DominicDesbiens** [comment]
Excellent point @AmauriC. Je suggère donc que l'on puisse avoir une option pour désactiver l'affichage du compte des cookies pour les services. Je vais le désactiver dans le code pour n'afficher que l'explication du service et les petits liens pointant vers les pages d'infos.

Merci pour vos retours.

PS : même si le compte de cookies n'est pas fait pour les partenaires de Adsense, aucun ne dépose les cookies lorsqu'on désactive Google Adsense (je vérifie le tout avec un outil de Nirsoft). C'est avec cet outil que j'ai constaté l'efficacité de ce script qui est de loin le meilleur que j'ai testé pour la mise en conformité de mon blogue (la plupart des autres scripts ne font qu'afficher une notification avec un simple bouton Accepter).

D'ailleurs, je me suis longuement questionné avant de mettre en place tarteaucitron sur mon site car je vois que de très nombreux sites d'infos et blogues français (et depuis peu québécois) ne font qu'afficher une notice sans possibilité de réglage des cookies (la notice dirige vers la politique de confidentialité qui montre comment désactiver les cookies dans son navigateur). Est-ce que le RGPD a une certaine « souplesse » à ce sujet ?!

**Comment by eldk** [comment]
Bonjour,

>Je parle notamment du cookie doubleclick.net
Pour le cookie doubleclick, il est possible d'empêcher sa lecture - même s'il a été implanté sur un autre site : https://github.com/AmauriC/tarteaucitron.js/issues/85

C'était la méthode indiquée dans une doc de la CNIL mais pour les utilisateurs de DFP. https://www.cnil.fr/fr/solutions-pour-la-publicite

Pour les autres cookies de suivi publicitaire, Google, dans la page de réglage de ses préférences (https://www.google.com/ads/preferences/) donne une solution d'optout (lien en bas à gauche) pour d'autres régies (http://www.youronlinechoices.com/fr/controler-ses-cookies). Cette solution doit-être exécutée, sur chaque appareil et chaque navigateur.

Cordialement,
Eric

**Comment by DominicDesbiens** [comment]
@eldk merci. Ce sont effectivement des solutions à proposer aux visiteurs de son site, je l'ai fait d'ailleurs dans ma politique de confidentialité récemment mise en ligne. C'est d'ailleurs la solution utilisée par de nombreux sites Internet (source du doute que j'ai eu au sujet de l'application du règlement) : ces sites affichent un bandeau et dirigent vers leur politique de confidentialité où ils proposent des façons de bloquer les cookies dans son navigateur. Or ce que dit le RGPD c'est qu'il faut donner le choix aux visiteurs de désactiver les cookies avant même qu'ils ne s'enregistrent dans leur appareil.

C'est une sacrée galère ce règlement (particulièrement sur la question des cookies) surtout lorsqu'on est à l'extérieur de l'Union européenne (je suis du Québec au Canada).

**Comment by eldk** [comment]
>Or ce que dit le RGPD c'est qu'il faut donner le choix aux visiteurs de désactiver les cookies avant même qu'ils ne s'enregistrent dans leur appareil.

Je pense que cela veut dire que l'acceptation implicite par "scroll" et par navigation (clic sur un lien et affichage d'une seconde page) doit être désactivée : 

```
"highPrivacy": false,
```
Ne serait donc plus autorisé ?

Cordialement,

Eric

**Comment by DominicDesbiens** [comment]
C'est là la confusion...

Article intéressant publié en avril dernier qui en parle : [https://www.avossitespros.com/bandeau-cookie-conforme-rgpd/](url)

**Comment by eldk** [comment]
Merci,
sur https://www.avossitespros.com , le scroll n'y est pas, par contre, le consentement par poursuite de la navigation est en place.

**Comment by fadupla** [comment]
Bonjour @all,

@DominicDesbiens merci pour ton lien. Dans cet article il est en effet écrit :

    [...] la poursuite de la navigation (changement de page, le défilement [scroll] dans la page ne suffit pas)

Il faudrait un lien vers texte légal qui exprime cette notion d'action de consentement pour conclure si oui ou non un scroll de page est un consentement (car on peut lire le contraire de l'article que tu donnes malheureusement sur d'autres sites).

**Comment by eldk** [comment]
Bonjour,

Je suis aller voir comment cela était implémenté sur le site de la [CNIL](https://www.cnil.fr/) : 

- Pas de consentement par scroll
- La poursuite de la navigation donne consentement pour tous les cookies.

Cordialement,
Eric

**Comment by DominicDesbiens** [comment]
Oui c'est exact. C'est d'ailleurs par le site de la CNIL que j'ai découvert tarteaucitron. Le changement de page agit à titre de consentement. Mais cela ne veut pas dire que l'on est 100 % légal. Avoir un bandeau qui notifie la présence de cookies et qui les active par défaut quand on change de page ne signifie pas que l'on est conforme au RGPD. Il faut d'abord et avant tout que le visiteur ait eu la possibilité de désactiver les cookies avant de poursuivre sa navigation.

@fadupla j'ai lu des dizaines de pages d'infos à ce sujet et c'est là le problème, beaucoup se contredisent. Je pense que le paragraphe sur les cookies dans le document du RGPD aurait dû être plus explicite à ce sujet.
